### PR TITLE
fix word prediction not being loaded for swiss french and dvorak layout

### LINKED ISSUE #100: Keyboard help desactivated when switching french-swiss keyboard to emoji and back to french-swiss keyboard
Little annoying bug found. Tested with app "Notes" or "Teleports" on N5, MX4

**Steps to reproduce**
1) Display language is "English UK"
2) Enable 3 keyboard layouts : EN, FR-CH, EMOJI
3) Spell checking, auto Capitalization, auto Punctuation are set to ON.

With the "earth symbol", select FR-CH -> word suggestion bar is visible:
- Start typing ".", then "space" -> keyboard is uppercase automatically
- Type "F", then "space" 2 times -> "." appears automatically

Select EMOJI -> word suggestion bar is not visible (that's ok)
- type some emoji

Select FR-CH -> word suggestion bar is NOT visible (weird)
- Start typing ".", then "space" -> keyboard is NOT uppercase automatically
- Type "F", then "space" 2 times -> "." does NOT appears automatically

Select EN, word suggestion bar is visible again, auto Cap and auto Punct work again
Select FR-CH, word suggestion bar is visible again, auto Cap and auto Punct work again
Then loop to "select EMOJI" to continue reproducing...

**Expected behavior**
Switch from FR-CH to EMOJI
Type some EMOJI
Switch back to FR-CH, word suggestion bar is still visible , auto Capitalization, auto Punctuation work

**Actual behavior**
Word suggestion bar, auto Cap and auto Punct are disabled when switching from FR-CH to EMOJI and back to FR-CH
Switching OFF/ON Word suggestion bar, auto Cap and auto Punct in "Language & Text" setting  has no effect.
Switch to EN and back to FR-CH makes Word suggestion bar, auto Cap and auto Punct work again.

BR
wilfridd

### LINKED ISSUE #132: Text prediction, autocorrection and the like not working automatically with swiss-french keyboard layout
- Device : FP2
- Channel : Stable (OTA-12)

### Steps to reproduce

1. In the system settings, set the keyboard layout to swiss-french only
2. Reboot the phone (optional, but just to be extra sure)
3. Open any app that makes you write text (Messages, Notes, TELEports)
4. Write something in french such as "Bonjour"

### Expected behaviour

Autocorrection, text prediction and such features should work.

### Actual behaviour

I can write as much text as I like, but the text checking features don't work.
**Workaround** : I switched to french keyboard and to swiss-french again, and then the text checking and autocorrecting features do work.

I am the one that "coded" the swiss-french keyboard layout back in 2018 (thank you very much for implementing it !). I just copied the files from the french keyboard and changed the keys to match my keyboard, expecting this to produce a working keybord out of the box. Hopefully the workaround works, but if would be pretty cool if the swiss-french keybord could be linked to the french dictionary from start. So what did I miss here ?

### Feature request

I see that the language is "Swiss-French". Would it be possible to translate the text to "Français (Suisse)" for the next keyboard ?

### PR DESCRIPTION
**Pull request by lduboeuf**
fixes https://github.com/ubports/keyboard-component/issues/132
fixes #100

### CODE REVIEW

**Comment by Fuseteam** [comment]
i'm not sure this is the way to go the db file is probably a copy of the french db file

**Comment by lduboeuf** [comment]
yes probably it is a copy from fr, if you find another way ;-).

**Comment by Fuseteam** [comment]
can we maybe symlink it?
i was talking with dalton about separating layout from language.
my pr for emoji is one way we could do just that. what i still haven't gotten to is getting the layouts visible in setting and how switching layouts should work in the keyboard(this is also the reason that can't be merged as there would be two almost identical emoji layouts in the tree)

**Comment by lduboeuf** [comment]
yes symlink should work. But is the db read-only or there can have idioms from the sub lang that can land there at runtime ?

**Comment by Fuseteam** [comment]
i don't think we have any mechanism to add idioms to the db atm

**Comment by lduboeuf** [comment]
So i give a try  by removing unnecessary files for fr-ch and allow to load the prefix database

**Comment by Fuseteam** [comment]
Hmm interesting this looks like it should rely on the french dictionary now.
Looking at the western support section it seems to imply the dvorak layout should just work, but iirc it had the same issue. Guess we should test if this works

**Comment by lduboeuf** [comment]
So to give info about this PR, currently, when switching from Emoji keyboard, the french dictionnary is not loaded when switching for fr-ch. I bet it is the same for en-dvorak

**Comment by Fuseteam** [comment]
confirmed it is the same for en@dv

**Comment by lduboeuf** [comment]
@Fuseteam so when switching from emoji to frznch swiss keyboard is working now;-).Sadly it still does'nt work for the english dvorak case, maybe just because the plugin is missing ( see src.pro INSTALLS in fr-ch)

I need to clean a bit this PR before marking it  ready for review

**Comment by lduboeuf** [comment]
@Fuseteam it works ;-), see https://github.com/ubports/keyboard-component/pull/134/commits/cbb88bfe938fbb87a6a7ac9fc62deec6ee83574e

**Comment by Fuseteam** [comment]
it is? cool thanks for the heads up @lduboeuf i'll check it out

**Comment by Fuseteam** [comment]
@lduboeuf i have tested this locally, feels like it needs a bit more than that commit, will continue testing it

it might useful to change this line: https://github.com/ubports/keyboard-component/blob/cbb88bfe938fbb87a6a7ac9fc62deec6ee83574e/plugins/fr-ch/src/src.pro#L23 to just "fr" then it should just read the db from the french layout

**Comment by lduboeuf** [comment]
@Fuseteam humm, haven't tested it, but if i read well, putting the plugin lib to "fr" will overwrite the existing one from the "fr", that is not what we would want. And if the lib isn't present the plugin will not be loaded ( that was the case for en@dv)

**Comment by Fuseteam** [comment]
@lduboeuf it does? i thought i did it the other way around hmmm. as far i've seen the db is read from that path

**Comment by lduboeuf** [comment]
> @lduboeuf it does? i thought i did it the other way around hmmm. as far i've seen the db is read from that path

Yes by default, but in my PR you will see a code addition to try to find the db also from the "root" language

**Comment by Fuseteam** [comment]
ah true, hmmm it doesn't seems to overwrite anything on my side when i do that, but then again it also doesn't fix it, hmmm

**Comment by Fuseteam** [comment]
btw @lduboeuf what does this line do?
https://github.com/ubports/keyboard-component/pull/134/files#diff-a258a48257ccfe094491277b334da558d8e34c6d55e959c8ca6212edfa7370e9

**Comment by lduboeuf** [comment]
> btw @lduboeuf what does this line do?
> https://github.com/ubports/keyboard-component/pull/134/files#diff-a258a48257ccfe094491277b334da558d8e34c6d55e959c8ca6212edfa7370e9

oh yes that one i can't remember sorry 

Edit: Or was it to avoid infinite looping

**Comment by Fuseteam** [comment]
> Or was it to avoid infinite looping

i have no idea :joy:

**Comment by lduboeuf** [comment]
@Fuseteam  have you tested this PR ? , it works on my side

**Comment by Fuseteam** [comment]
@lduboeuf i have tested commits from this pr in isolation so far i'll test the pr itself and let ya know

**Comment by Fuseteam** [comment]
hmm doesn't seem to work on my end when i install with ubports-qa, but i'm also stacking PRs atm

**Comment by wilfridd** [comment]
(https://github.com/ubports/keyboard-component/issues/100)
keyboard helpers fixed with swiss french layout
Thanks!

File: plugins/westernsupport/spellpredictworker.cpp [hunk 1]
```diff
@@ -88,10 +88,28 @@ void SpellPredictWorker::setLanguage(QString locale, QString pluginPath)
 {
     // locale for secondary layouts I.E., dvorak will be formatted as locale@layout
     // in this case we want to drop the layout portion
```
Comment (UniversalSuperBox) [inline plugins/westernsupport/spellpredictworker.cpp:90]: It appears that this comment has become out of date now that this code handles `-` as much as it handles `@`.

**Review by UniversalSuperBox — CHANGES_REQUESTED** [review]

File: plugins/westernsupport/spellpredictworker.cpp (hunk 1 above)
Comment (lduboeuf) [reply plugins/westernsupport/spellpredictworker.cpp:90]: humm and i can't read easily my code, need to comment and/or change variable names

**Review by UniversalSuperBox — APPROVED** [review]

**Pull request merged**